java/vector/src/test/java/org/apache/arrow/vector/TestDenseUnionVector.java

```diff
@@ -99,6 +99,44 @@ public void testDenseUnionVector() throws Exception {
     }
   }
 
+  @Test
+  public void testSetOffset() {
+    try (DenseUnionVector duv = DenseUnionVector.empty("foo", allocator)) {
+      duv.allocateNew();
+      byte i32TypeId = duv.registerNewTypeId(Field.notNullable("i32", MinorType.INT.getType()));
+      byte f64TypeId = duv.registerNewTypeId(Field.notNullable("f64", MinorType.FLOAT8.getType()));
+
+      IntVector i32Vector = ((IntVector) duv.addVector(i32TypeId, new IntVector("i32", allocator)));
```

Comment:
Should we use try-with-resources?

Reply:
No, the union vector will close it itself.